memcached.h

```diff
@@ -725,11 +751,10 @@ struct conn {
     /* data for the swallow state */
     int    sbytes;    /* how many bytes to swallow */
 
+    int io_queues_submitted; /* see notes on io_queue_t */
+    io_queue_t io_queues[3]; /* set of deferred IO queues. */
```

Comment:
short answer: because the possible queues are extstore, proxy, and the final must be NULL.

longer answer: eventually this could be used by a DSO system hook, so at that point it would change to either a realloc as hooks get added or a linked list. Hardcoding was simplest for now.

Reply:
Hi @dormando  - Few questions not specfic to this change-set, but came across while reviewing -

How are we protected by parallel extstore threads writing to the same pages? 
I see the following mutex initialized, but not used anywhere in the code. 
pthread_mutex_init(&e->pages[i].mutex, NULL);

typedef struct _store_page {
    pthread_mutex_t mutex; /* Need to be held for most operations */          <<<<<<
...
} store_page;

Reply:
that mutex is hit in a bunch of places in extstore.c, usually called like `p->mutex`

Reply:
Oh yes, that makes sense. Thank you.